base_layer/core/src/chain_storage/error.rs

```diff
@@ -54,7 +51,7 @@ pub enum ChainStorageError {
     #[error("A parameter to the request was invalid")]
     InvalidQuery(String),
     #[error("The requested value '{0}' was not found in the database")]
-    ValueNotFound(DbKey),
+    ValueNotFound(String),
```

Comment:
Why was this changed? The `{0}` calls DbKey::to_string() internally and the `ValueNotFound` error is only applicable for variants of `DbKey`.

Reply:
Though this might come up. The problem is, I am not searching them via the usual keys. 
For example DB::key assumes you will always search a kernel via hash, we dont here, we search via the excess_sig. 

So if I cant find the error I have to create ValueNotFound(Kernel::hash), but I dont have the hash. So I cant create a value not found

Reply:
Having the DbKey gives two pieces of information 1. The entity we are searching for and 2. The value we are searching for. It actually also tells us that we are searching for it by the id. Maybe we should expand the ValueNotFound to take these params: Table/Entity name (string), Field being searched (string), Value (string), and then potentially the cause as an addition param?

Reply:
I will create a struct for this all.

Reply:
Why not individual params? Check out the InvalidHeader error here: https://docs.rs/thiserror/1.0.20/thiserror/